handler/src/main/java/io/netty/handler/ssl/SslContextBuilder.java

```diff
@@ -629,13 +634,31 @@ public SslContextBuilder secureRandom(SecureRandom secureRandom) {
      *     Java Security Standard Names</a> for a list of supported algorithms.
      *
      * @param algorithm either {@code "HTTPS"}, {@code "LDAPS"}, or {@code null} (disables hostname verification).
-     * @see javax.net.ssl.SSLParameters#setEndpointIdentificationAlgorithm(String)
+     * @see SSLParameters#setEndpointIdentificationAlgorithm(String)
      */
     public SslContextBuilder endpointIdentificationAlgorithm(String algorithm) {
         endpointIdentificationAlgorithm = algorithm;
         return this;
     }
 
+    /**
+     * Add the given server name indication to this client context. This will cause the client to include a
+     * Server Name Indication extension with its {@code ClientHello} message, as per
+     * <a href="https://datatracker.ietf.org/doc/html/rfc6066#section-3">RFC 6066 section 3</a>.
+     * <p>
+     * Note that only one name per name type can be included in the message.
+     * Currently, only the {@link SNIHostName} type is supported.
```

Comment:
Should we do a type check ?

Reply:
We type check in the engines, but since there's only one type supported as per the standard, and I don't really see that changing for a while, we might as well pull the check forward.